### PR TITLE
Enhance highlights tree style

### LINKED ISSUE #2635: Support finer-grained control over treegrid row shading

### PR DESCRIPTION
**Pull request by petradish**
Resolves #2635 to differentiate between various levels of a tree grid. Supports up to 6 levels in light and dark mode.
Adds new Colors and Colors w/Highlights tree styles.

Toolbox PR to add new tree styles to GridOptions.js: https://github.com/xh/toolbox/pull/532

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [N/A] Added CHANGELOG entry, or determined not required.
- [N/A] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [N/A] Updated doc comments / prop-types, or determined not required.
- [N/A] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by petradish** [comment]
Screen Shot 2021-09-07 at 4 44 04 PM
Screen Shot 2021-09-07 at 4 44 23 PM
Screen Shot 2021-09-07 at 4 44 43 PM
Screen Shot 2021-09-07 at 4 44 54 PM

**Comment by lbwexler** [comment]
Petra, I converted this to a DRAFT (and removed WIP from the name) according to our new standards.

But let us know if you have updated this to the latest and greatest and want to try and push this over the top.  I think its a clear improvement.  We can always tweak the colors and  the API change seems quite minimal.

I guess one question to answer is, if an app  did want to tweak the colors, how straightforward would that be?

File: styles/vars.scss [hunk 1]
```diff
@@ -422,14 +433,20 @@ body {
...
+    --xh-grid-bg-hover: var(--grid-bg-hover, hsl(200, 50%, 22%));
     --xh-grid-bg-odd: var(--grid-bg-odd, hsl(200, 26%, 11%));
     --xh-grid-border-color: var(--grid-border-color, hsl(186, 11%, 18%));
     --xh-grid-cell-change-bg-highlight: var(--grid-cell-bg-highlight, #{mc-trans('orange', '800', 0.8)});
     --xh-grid-group-bg: var(--grid-group-bg, #{mc('blue-grey', '800')});
     --xh-grid-group-border-color: var(--grid-group-border-color, #{mc('blue-grey', '700')});
-    --xh-grid-tree-group-bg: var(--grid-tree-group-bg, hsl(200, 50%, 12%));
+    --xh-grid-tree-group-bg: var(--grid-tree-group-bg, hsl(200, 50%, 16%));
     --xh-grid-total-row-bg: var(--grid-total-row-bg, var(--xh-grid-tree-group-bg));
+    --xh-grid-tree-group-color-level-0: var(--grid-tree-group-color-level-0, hsl(205, 65%, 15%));
```
Comment (petradish) [inline styles/vars.scss:444]: These hsl colors look better than what I could create using Material colors w/ a darken function. Could take a second pass at using material colors for consistency with light theme

File: cmp/grid/Grid.scss [hunk 2]
```diff
@@ -211,12 +211,16 @@
...
   // rows at the same level.
-  @for $i from 0 to 20 {
+  @for $i from 0 to 10 {
     &.xh-grid--tree-style-highlights.xh-grid--max-depth-#{$i} {
-      .ag-row:not(.ag-row-level-#{$i}) {
-        &:not(.ag-row-selected),
-        &.ag-row.ag-row-odd:not(.ag-full-width-row):not(.ag-row-selected) {
-          background-color: var(--xh-grid-tree-group-bg);
+      @for $level from 0 to 9 {
+        $l: 94 + ($level / ($i + 1) * 6);
```
Comment (amcclain) [inline cmp/grid/Grid.scss:217]: I see the following warning when compiling:
```

    ╷
217 │         $l: 94 + ($level / ($i + 1) * 6);
    │                   ^^^^^^^^^^^^^^^^^
    ╵
    ../../hoist-react/cmp/grid/Grid.scss 217:19  root stylesheet

: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($level, $i + 1)
```

File: styles/vars.scss (hunk 1 above)
Comment (amcclain) [reply styles/vars.scss:444]: I'm not especially loyal to the material colors, especially in dark mode.  (I wonder if there is a different set of palettes they make available specifically for dark themes?)

In some ways the material design color function ends up obscuring the underlying color / makes it a bit more difficult to reason about for a dev who wants to find some HSL variant or generally re-use the color w/o necessarily importing or using the mc functions.  We're clearly in an inconsistent state at the moment and I am not proposing we try to resolve everything one way or another - just saying I wouldn't want use of material colors to compromise the end result if we are happier with HSL.

**Review by amcclain — COMMENTED** [review]
Eager to get this merged!

+ I am running the HR and TB branches and don't see the group borders at all when selecting those options - are you?  (I don't see how this change would have broken those, but they are working on toolbox-dev.)
+ To really appreciate this change, we need at least one (or ideally two) levels of additional grouping in our Toolbox example. Could you please look to see if you can come up with something? I'd suggest Long/Short and Profit/Loss - two easy dimensions based on the numeric data that doesn't require faking anything new.

**Comment by petradish** [comment]
I am seeing the group borders on my local machine for those 2 w/Border options in the example. I'm not sure why you can't!

File: styles/vars.scss (hunk 1 above)
Comment (petradish) [reply styles/vars.scss:439]: Updated to use HSL colors in light theme

**Pull request merged**